.travis.yml

```diff
@@ -8,110 +8,157 @@ os: linux
 language: cpp
 env:
   global:
-    - PYTHON_VERSION="2.7.15"
+    - PYTHON2_VERSION="2.7.15"
+    - PYTHON3_VERSION="3.6.7"  # "3.7.1" after #29326 is fixed
+    - PYTHON2_CACHE=$HOME/.ccache/py${PYTHON2_VERSION}
+    - PYTHON3_CACHE=$HOME/.ccache/py${PYTHON3_VERSION}
+cache:
+  ccache: true
+  directories:
+    - ${PYTHON2_CACHE}
+    - ${PYTHON3_CACHE}
 jobs:
   include:
     - stage: "Compile"
-      name: "Compile V8"
-      cache: ccache
+      name: "Compile V8 (py2)"
       addons:
         apt:
+          update: true
```

Comment:
Should the `update: true` lines should be there in the final result or should that last commit (that adds them) be omitted when landing? I'm guessing they should be there but I'm asking anyway because the commit message suggests it might be temporary--"Force a retest"--and the Travis docs mildly discourage its use.

Reply:
I think they can stay in.  I believe that the Travis docs are discouraging __upgrade__vs. __update__.